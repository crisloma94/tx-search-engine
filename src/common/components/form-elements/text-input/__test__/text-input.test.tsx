import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Theme from '../../../../../styles/theme';
import TextInput, { TextInputProps } from '../text-input.component';
import { store } from '../../../../../core/store/store';

describe('Text input component test suite', () => {
  const getComponent = (props: TextInputProps) => (
    <Provider store={store}>
      <Theme>
        <TextInput {...props} />
      </Theme>
    </Provider>
  );
  const setup = (props: TextInputProps) => render(getComponent(props));

  const onClickMock = jest.fn();
  const handleChangeMock = jest.fn();
  const defaultProps = {
    placeholder: 'Text input placeholder',
    onClick: onClickMock,
    handleChange: handleChangeMock,
    value: 'Text input value',
    name: 'Text input name',
  };

  it('Should match snapshot', () => {
    const { container } = setup(defaultProps);
    expect(container).toMatchSnapshot();
  });

  it('Should call the onClick function when clicked', async () => {
    setup(defaultProps);
    fireEvent.click(screen.getByRole('textbox'));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('Should call the handleChange function when clicked', async () => {
    setup(defaultProps);
    fireEvent.change(screen.getByRole('textbox'), {
      target: { value: 'test' },
    });
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
