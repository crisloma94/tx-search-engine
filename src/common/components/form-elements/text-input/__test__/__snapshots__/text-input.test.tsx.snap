// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text input component test suite Should match snapshot 1`] = `
<div>
  <input
    class="sc-bgqQcB gWCqZe"
    name="Text input name"
    placeholder="Text input placeholder"
    type="text"
    value="Text input value"
  />
</div>
`;
